### ISSUE TITLE
Erroring on aspect ratio based media queries

### ISSUE THREAD

**Issue by Dan503** [jonathantneal/postcss-unmq#2]
I have downloaded the latest version of unMQ just to be sure this isn't a version issue. 

When I use this css, I get a `Cannot read property '1' of null` error

`````css
@media screen and (max-aspect-ratio: 2/1){
	* { background: red; }
}
`````

I realise that it's a bit hard to tell if a ratio based media query should be output into a version of the style sheet with the media queries stripped out. You will probably need to provide some settings like, strip all out, leave all in, leave them in if they are above/below a defined ratio. I think by default they should be stripped out to keep things simple though.

Or is this an issue with css-mediaquery? https://github.com/ericf/css-mediaquery 

This is the stack trace:

`````````````
events.js:141
      throw er; // Unhandled 'error' event
      ^
 TypeError: Cannot read property '1' of null
    at C:\xxxxxx\node_modules\css-mediaquery\index.js:105:36
    at Array.map (native)
    at C:\xxxxxx\node_modules\css-mediaquery\index.js:103:42
    at Array.map (native)
    at parseQuery (C:\xxxxxx\node_modules\css-mediaquery\index.js:88:34)
    at Object.matchQuery [as match] (C:\xxxxxx\node_modules\css-mediaquery\index.js:21:12)
    at C:\xxxxxx\node_modules\postcss-unmq\index.js:34:19
    at C:\xxxxxx\node_modules\postcss\lib\container.js:308:28
    at C:\xxxxxx\node_modules\postcss\lib\container.js:148:26
    at Root.each (C:\xxxxxx\node_modules\postcss\lib\container.js:114:22)
`````````````

(PS. Don't forget that there are also the `device-aspect-ratio` and `orientation` media query types)

**Comment by jonathantneal** [comment]
Could you make a repo demonstrating this failure? I’ve added a [test for `max-aspect-ratio: 2/1`](https://github.com/jonathantneal/postcss-unmq/blob/master/test/fixtures/aspect-ratio.css) to the repository, and it works. You may notice I slightly altered the test from yours, but I assure you I tried it with your exact selector and declaration as well. If you’re able to provide an example, I’ll gladly re-open the issue. According to your error, it seems something ran afoul with `css-mediaquery` on [line 105](https://github.com/ericf/css-mediaquery/blob/master/index.js#L105). And yes, that line comes up as an empty one for me, too, probably because their repository is [not in sync with what they’ve published to npm](https://github.com/ericf/css-mediaquery/issues/16). Unfortunately, it also looks like [development stopped 2 years ago](https://github.com/ericf/css-mediaquery/commits/master).

I’ll gladly fork their repo, make a PR, request collaborator status or create a local version of that library for this plugin in order to repair this issue if you’re able to narrow it down a bit further. Thanks for providing as many details as you could. I’m sorry I was unable to figure it out from just that log.

**Comment by Dan503** [comment]
I discovered the error while I was working on my [YeoSword](https://github.com/Dan503/YeoSword) project.

You can replicate the error by checking out a copy of [YeoSword](https://github.com/Dan503/YeoSword) then in the `src/_0_styles/main.scss` file, add the aspect-ratio media query. Point a command prompt at the root repository folder, do an `npm install` and then run `gulp sass`.

You should see an error that looks like this:
![error message](http://grabilla.com/07107-1eb9c1f8-f075-4501-9ac5-22ad43b5316c.png)

Open the `gulp/tasks/sass.js` file. On line 105 you will see the `postcss.unmq({ width: bp.page + 10 })` line there. Commenting out that line removes the error.

In order to get the full stack trace error that I posted above, you will need to remove `.pipe(plugins.plumber())` on line 98.

**Comment by Dan503** [comment]
oh and the compiled css gets output into the `tmp/path/to/base/assets/styles` folder. 

The affected css is the `noMQs-main.css` file.

**Comment by jonathantneal** [comment]
What is the value of `bp.page`? I’ve tried various values (string, null, undefined, null, NaN) and cannot trigger this error.

**Comment by Dan503** [comment]
It's just the number `1200`. `bp` is imported from this file:

https://github.com/Dan503/YeoSword/blob/master/src/_0_scripts/_config/bp_break-points.js

I'll need to try doing a fresh install of YeoSword to make sure I can still replicate the issue based on the steps I gave.

**Comment by Dan503** [comment]
I use a windows 10 PC if that makes any difference.

**Comment by jonathantneal** [comment]
`width: 1200 + 10` worked for me. I was wondering if Sass was trying to work out the math for you and turn `2/1` into `2`, but even trying `max-aspect-ratio: 2` failed to break this plugin.

**Comment by Dan503** [comment]
I just did a fresh install of YeoSword and followed the directions I gave. 

The error **_did_** occur. I'm not sure why you aren't able to replicate it :(

The steps I took:

1. Downloaded the [YeoSword zip file](https://github.com/Dan503/YeoSword/archive/master.zip) and saved the contents into a folder
2. Pointed a command prompt at the folder
3. ran `npm install`
4. opened `src/_0_styles/main.scss`
5. added `@media screen and (max-aspect-ratio: 2/1){	* { background: red; } }` to the bottom of the file and saved
6. ran `gulp sass`

That should cause the error to occur.

In the process, I discovered that the error prevents the noMQs css file from ever being generated.
